BD-3260 Snowflake Data Retention (#7681)
braze-inc/braze-docs: _docs/_partners/data_and_infrastructure_agility/data_warehouses/snowflake/data_retention.md

---
nav_title: "Data Retention"
article_title: Snowflake Data Retention
page_order: 3
description: "This page covers how to retain full events data when the Braze retention policy is applied."
page_type: partner
search_tag: Partner
---

# Snowflake data retention

> > Braze anonymizes (removes personally identifiable information (PII)) from all events data that is older than two years old. If you use Snowflake data sharing, you may choose to retain the full events data in your environment by storing a copy in your Snowflake account before the retention policy is applied.

Reviewer: ```suggestion
> Braze anonymizes—removes personally identifiable information (PII)—from all events data that is older than two years old. If you use Snowflake data sharing, you may choose to retain the full events data in your environment by storing a copy in your Snowflake account before the retention policy is applied.
```

This page presents two ways you can retain non-anonymized data: 

- Copy your data to another Snowflake database
- Unload your data to a stage

{% alert warning %}

> Braze automatically anonymizes events data for users that are deleted from Braze, as described in [Data Protection Technical Assistance]({{site.baseurl}}/dp-technical-assistance/). Any data copied outside of the shared database will not be included in this process, as it’s no longer managed by Braze. 

Reviewer 2: The second sentence of this is very important, but for some reason it's not showing up in the warning when I look at the preview deployment

Reviewer 2: <img width="993" alt="image" src="https://github.com/user-attachments/assets/e405996b-3f31-4a36-b96f-d8598e293ca5">

> Braze automatically anonymizes events data for users that are deleted from Braze, as described in [Data Protection Technical Assistance]({{site.baseurl}}/dp-technical-assistance/). Any data copied outside of the shared database will not be included in this process, as it’s no longer managed by Braze. 

Reviewer: ```suggestion
Braze automatically anonymizes events data for users that are deleted from Braze, as described in [Data Protection Technical Assistance]({{site.baseurl}}/dp-technical-assistance/). Any data copied outside of the shared database will not be included in this process, as Braze no longer manages it. 
```

            result.append([table_name, False, str(e)])
    
    -- Return the results
    return session.create_dataframe(result, schema=['TABLE_NAME', 'SUCCESS', 'INFO'])
$$;
```
{% endraw %}

{: start="2"}
2. Run one of the below commands in your Snowflake account to execute the procedure.

{% tabs %}
{% tab Default %}

By default, the procedure will back up data older than two years for all `USERS_*` event types. 

{% raw %}
```sql
-- Copy all the rows that are two years or older in all the 'USERS_*' tables 
-- from 'SOURCE_DB'.'SOURCE_SCHEMA' to 'DEST_DB'.'DEST_SCHEMA'

CALL COPY_BRAZE_SHARE('SOURCE_DB', 'SOURCE_SCHEMA', 'DEST_DB', 'DEST_SCHEMA')
```
{% endraw %}
{% endtab %}
{% tab Filtered %}

Specify a filter to choose what age data to back up, and specify a table name filter to back up only selected events tables. 

{% raw %}
```sql
-- Copy all the rows that are one year or older in all the 'USERS_BEHAVIORS_*' tables
-- from 'SOURCE_DB'.'SOURCE_SCHEMA' to 'DEST_DB'.'DEST_SCHEMA'

CALL COPY_BRAZE_SHARE('SOURCE_DB', 'SOURCE_SCHEMA', 'DEST_DB', 'DEST_SCHEMA', DATEADD(year, -1, CURRENT_DATE()), 'USERS_BEHAVIORS_%')
```
{% endraw %}
{% endtab %}
{% endtabs %}

{% alert note %}

> Repeat running of the procedure won't create duplicate records, as the procedure will check the most recent `SF_CREATED_AT` and only back up data newer than that. 

Reviewer: ```suggestion
Repeatedly running the procedure won't create duplicate records since this procedure checks the most recent `SF_CREATED_AT` and only backs up data newer than that. 
```
